lib-sql/functions/frozen-db-functions.sql

```diff
@@ -0,0 +1,131 @@
+-- These functions query unified tables instead of partitions,
+-- which allows TIGER data imports on frozen databases.
+
+-- TODO: add reinstatiation of partition functions if needed
+DROP FUNCTION IF EXISTS getNearestNamedRoadPlaceId(integer, geometry, jsonb) CASCADE;
+DROP FUNCTION IF EXISTS find_road_with_postcode(integer, text) CASCADE;
+DROP FUNCTION IF EXISTS getNearestRoadPlaceId(integer, geometry) CASCADE;
+DROP FUNCTION IF EXISTS getAddressName(integer, geometry) CASCADE;
+DROP FUNCTION IF EXISTS getNearestParallelRoadFeature(integer, geometry) CASCADE;
+
+
+CREATE FUNCTION getNearestParallelRoadFeature(in_partition INTEGER,
+                                             line GEOMETRY)
+  RETURNS BIGINT
+  AS $$
+DECLARE
+  r RECORD;
+  search_diameter FLOAT;
+  p1 GEOMETRY;
+  p2 GEOMETRY;
+  p3 GEOMETRY;
+BEGIN
+
+  IF ST_GeometryType(line) not in ('ST_LineString') THEN
+    RETURN NULL;
+  END IF;
+
+  p1 := ST_LineInterpolatePoint(line, 0);
+  p2 := ST_LineInterpolatePoint(line, 0.5);
+  p3 := ST_LineInterpolatePoint(line, 1);
+
+  search_diameter := 0.0005;
+  WHILE search_diameter < 0.01 LOOP
+    FOR r IN
+      SELECT place_id FROM placex
+        WHERE ST_DWithin(line, geometry, search_diameter)
```

Comment:
You need to restrict to road feature here. This is done with: `rank_address between 26 and 27`.